application/src/main/java/org/togetherjava/tjbot/features/chatgpt/ChatGptService.java

```diff
@@ -96,12 +97,58 @@ public Optional<String> ask(String question, String context) {
             return Optional.empty();
         }
 
+        String instructions = "KEEP IT CONCISE, NOT MORE THAN 280 WORDS";
+        String questionWithContext = "context: Category %s on a Java Q&A discord server. %s %s"
+            .formatted(context, instructions, question);
+
+        return getMessageResponse(questionWithContext);
+    }
+
+    /**
+     * Prompt ChatGPT with code for it to format it.
+     *
+     * @param code the code to be formatted by ChatGPT. If code exceeds {@value MAX_CODE_LENGTH}
+     *        characters then this method returns an empty {@link Optional}
+     * @return an optional response from ChatGPT as a String
+     */
+    public Optional<String> formatCode(CharSequence code) {
+        if (isDisabled || code.length() > MAX_CODE_LENGTH) {
+            return Optional.empty();
+        }
+
+        String payload = String.format(
+                """
+                                If you happen to find any code in the container below, FORMAT \
+                                IT regardless of the programming language you find. MAKE IT HUMANLY READABLE. \
+                                If you don't find any, then your only answer should say empty. Output with no \
+                                introduction, no explanation, no ``` stuff, only code. Double check that \
+                                your response is correct. The code provided might not be readable.
+
+                                --- BEGIN CODE ---
+                                %s
+                                --- END CODE ---
+                        """,
+                code);
+
+        Optional<String> response = getMessageResponse(payload);
+
+        if (response.isEmpty() || response.get().equalsIgnoreCase("empty")) {
```

Comment:
Shouldn't just checking optional is empty enough here? Why prompt GPT to respond with "empty" 🧐?

Reply:
What prompt should we change it to so that it responds with literally no answer instead of instructing it to respond with "empty"? And even in that case, won't the OpenAI API return an empty string?

Reply:
why would it respond with no answer? if for some reason API fails we'll get empty optional. Was there any instance of it returning empty string?

We won't even need to check and directly do 
`return getMessageResponse(payload);`

Reply:
But what if the API receives non-valid code or code that's too long? It's instructed to respond with "empty" if any of the predefined conditions are not met in order to correctly decide whether to delegate the code formatting task to the original code or not.

Feel free to share any better methods that would solve this.

Reply:
Let me try and address some of the concerns

1. This will be applied by someone when they use context menu options for code action, it won't be automatically formatted.
2. For formatting there's no such thing as invalid code, it would apply the formatting rules automatically of a certain language based on input even if it's a diff langauges then what it interprets.
3. As for length, you're already checking length for code greater than 2k chars to return optional at start of the method.

the method is itself is solid, but this if check ` if (response.isEmpty() || response.get().equalsIgnoreCase("empty"))` seems unnecessary imo

you can just do `return getMessageResponse(payload)` instead.

and also can safely get rid of prompt for "empty" string as well

Reply:
i kinda agree, just let gpt do whatever it can,  and if it fails  to  respond  then only use the other formatter